atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepMetrics.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.sweep;
+
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SlidingTimeWindowReservoir;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+
+final class SweepMetrics {
+    private final MetricRegistry metricRegistry;
+
+    public static SweepMetrics create() {
+        SweepMetrics sweepMetrics = new SweepMetrics();
+        sweepMetrics.registerAggregateMetrics();
+        return sweepMetrics;
+    }
+
+    private void registerAggregateMetrics() {
+        SlidingTimeWindowReservoir reservoir = new SlidingTimeWindowReservoir(1, TimeUnit.DAYS);
```

Comment:
Specifically I'm worried that under heavy sweep deletes, the daily and weekly metrics will consume significant memory. For example, doing 1 million updates to a `SlidingTimeWindowReservoir` will require ~84MB. A similar HdrHistogram backed reservoir (e.g. from Tritium) would take ~74KB.

![image](https://cloud.githubusercontent.com/assets/54594/23724003/3d1cee3e-0419-11e7-9bbe-076f68b8446d.png)